File: ECNU online judge/2006.py

ia = [1] * 1002000
i = 1
while(i <= 1000000):
	tmp = i
	sum = 0
	while(tmp > 0):
		sum = sum + (tmp % 10)
		tmp = int(tmp/10)
		print(tmp)
	sum = int(sum)
	ia[i+sum] = 0
	i = i+1
i = 1
while(i <= 1000000):
	if ia[i] == 1:
		print(i)
	i = i+1

File: ECNU online judge/2006.cpp
#include<bits/stdc++.h>
using namespace std;
const int maxn = 1000000+10;
bool is[maxn];
int ans[maxn], an;
int main(){
	memset(is, 1, sizeof(is));
	for (int i=1; i<=1000000; ++i){
		int sum = 0, tmp = i;
		while(tmp){
			sum += tmp%10;
			tmp /= 10;
		}
		is[sum+i] = false;
	}
	for (int i=1; i<=1000000; ++i)
		if (is[i])
			printf("%d\n", i);
	return 0;
}